testing/internal/e2e/tests/base/search_test.go

```diff
@@ -0,0 +1,197 @@
+		},
+	)
+	require.NoError(t, err)
+
+	// Set up a new org and project
+	boundary.AuthenticateAdminCli(t, ctx)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId)
+
+	// Create enough targets to overflow a single page.
+	// Use the API to make creation faster.
+	t.Log("Creating targets...")
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+	tClient := targets.NewClient(client)
+	targetPort, err := strconv.ParseInt(c.TargetPort, 10, 32)
+	require.NoError(t, err)
+	var targetIds []string
+	targetPrefix := "test-target"
+	for i := 0; i < c.MaxPageSize+1; i++ {
+		resp, err := tClient.Create(ctx, "tcp", newProjectId,
+			targets.WithName(targetPrefix+strconv.Itoa(i)),
+			targets.WithTcpTargetDefaultPort(uint32(targetPort)),
+			targets.WithAddress(c.TargetAddress),
+		)
+		require.NoError(t, err)
+		targetIds = append(targetIds, resp.Item.Id)
+	}
+
+	// List targets recursively.
+	// This requests data from the controller/database.
+	t.Log("Listing targets...")
+	output = e2e.RunCommand(ctx, "boundary", e2e.WithArgs("targets", "list", "-scope-id", newProjectId, "-format", "json"))
+	require.NoError(t, output.Err, string(output.Stderr))
+	var targetListResult targets.TargetListResult
+	err = json.Unmarshal(output.Stdout, &targetListResult)
+	require.NoError(t, err)
+	var listedIds []string
+	for _, item := range targetListResult.Items {
+		listedIds = append(listedIds, item.Id)
+	}
+	require.Equal(t, len(targetIds), len(listedIds))
+
+	err = backoff.RetryNotify(
+		func() error {
+			output = e2e.RunCommand(ctx, "boundary",
+				e2e.WithArgs(
+					"search",
+					"-resource", "targets",
+					"-format", "json",
+					"-query", fmt.Sprintf(`name %% "%s" and scope_id = "%s"`, targetPrefix, newProjectId),
+				),
+			)
+			if output.Err != nil {
+				return backoff.Permanent(errors.New(string(output.Stderr)))
+			}
+
+			var searchResult clientcache.SearchResult
+			err = json.Unmarshal(output.Stdout, &searchResult)
+			require.NoError(t, err)
+			if err != nil {
+				return backoff.Permanent(err)
+			}
+
+			targetCount := len(searchResult.Item.Targets)
+			if targetCount == 0 {
+				return errors.New("No targets are appearing in the search results")
+			}
+
+			t.Logf("Found %d target(s)", targetCount)
+			return nil
+		},
+		backoff.WithMaxRetries(backoff.NewConstantBackOff(3*time.Second), 5),
+		func(err error, td time.Duration) {
+			t.Logf("%s. Retrying...", err.Error())
+		},
+	)
+	require.NoError(t, err)
+
+	// Search for targets that contain the target prefix.
+	// This requests data from the client cache daemon.
+	// The force refresh option forces the client daemon to fetch new data
```

Comment:
Should this comment go above the retry above? It was a bit surprising to see it down here